src/parser.rs

```diff
@@ -0,0 +1,112 @@
+use crate::ast::{CommonExpr, Literal};
+use nom::branch::alt;
+use nom::bytes::complete::{tag, tag_no_case};
+use nom::character::complete::{char, digit1, one_of};
+use nom::combinator::{cut, map, opt, recognize, value, verify};
+use nom::error::{Error, ParseError};
+use nom::sequence::{pair, tuple};
+use nom::IResult;
+use nom::ParseTo;
+
+pub fn parse_float(inp: &str) -> IResult<&str, f64> {
```

Comment:
`nom` has a built-in parser for floats as well, but it does not match OData syntax exactly.